src/Elastic.Apm/Helpers/StacktraceHelper.cs

```diff
@@ -20,9 +21,26 @@ internal static class StacktraceHelper
 		/// <param name="frames">The stack frames to rewrite into APM stack traces</param>
 		/// <param name="logger">The logger to emit exceptions on should one occur</param>
 		/// <param name="dbgCapturingFor">Just for logging.</param>
+		/// <param name="configurationReader">
+		/// Config reader - this controls the collection of stack traces (e.g. limit on frames,
+		/// etc)
+		/// </param>
 		/// <returns>A prepared List that can be passed to the APM server</returns>
-		internal static List<CapturedStackFrame> GenerateApmStackTrace(StackFrame[] frames, IApmLogger logger, string dbgCapturingFor)
+		internal static List<CapturedStackFrame> GenerateApmStackTrace(StackFrame[] frames, IApmLogger logger,
+			IConfigurationReader configurationReader, string dbgCapturingFor
+		)
 		{
+			var stackTraceLimit = configurationReader.StackTraceLimit;
```

Comment:
Yeah, that's a good point. I changed this, we only read it only once. For now I think we are good with this, and we already have open issues for hot changeable configs, so this will probably evolve later.